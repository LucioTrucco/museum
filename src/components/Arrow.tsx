import React, { FC } from 'react';
import { ColorValue, View } from 'react-native';
import Svg, { Path } from 'react-native-svg';
import { colors } from '@/styles';

type Props = {
  color?: ColorValue;
  direction: 'up' | 'right' | 'down' | 'left'; // use object literals for stronger types
};

const DIRECTIONS = ['up', 'right', 'down', 'left'];

export const Arrow: FC<Props> = ({ direction, color = colors.silver }) => {
  const rotationAngle = (DIRECTIONS.indexOf(direction) * 90 + 180) % 360;
  return (
    <View
      style={{
        transform: [
          {
            rotate: `${rotationAngle}deg`,
          },
        ],
      }}>
      <Svg width={11} height={7} viewBox="0 0 11 7" fill="none">
        <Path
          d={`M1.068 
          1.329a.25.25 0 0 0 .01.353l4.25 4a.25.25 0 0 0 .343 0l4.25-4a.25.25 
          0 1 0-.342-.364L5.5 5.157l-4.079-3.84a.25.25 0 0 0-.353.012z`}
          fill={color}
          stroke={color}
          strokeWidth={2}
        />
      </Svg>
    </View>
  );
};
