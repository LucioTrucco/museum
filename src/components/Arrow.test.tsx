import React from 'react';
import { render } from '@testing-library/react-native';
import { Arrow } from './Arrow';

describe('tests => SectionTitle', () => {
  let component: any;

  beforeEach(() => {
    component = render(<Arrow direction="left" />);
  });

  it('should render Arrow', () => {
    expect(component.toJSON().props.style.transform[0].rotate).toEqual('90deg');
  });

  it('should match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
