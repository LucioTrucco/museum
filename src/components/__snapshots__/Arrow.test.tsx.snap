// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests => SectionTitle should match snapshot 1`] = `
<View
  style={
    {
      "transform": [
        {
          "rotate": "90deg",
        },
      ],
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="7"
    bbWidth="11"
    fill="none"
    focusable={false}
    height={7}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 7,
          "width": 11,
        },
      ]
    }
    vbHeight={7}
    vbWidth={11}
    width={11}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M1.068 
          1.329a.25.25 0 0 0 .01.353l4.25 4a.25.25 0 0 0 .343 0l4.25-4a.25.25 
          0 1 0-.342-.364L5.5 5.157l-4.079-3.84a.25.25 0 0 0-.353.012z"
        fill={
          {
            "payload": 4292796126,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
          ]
        }
        stroke={
          {
            "payload": 4292796126,
            "type": 0,
          }
        }
        strokeWidth="2"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
